appendix-basic65.tex

```diff
@@ -187,30 +192,57 @@ \subsection{BASIC 65 Constants}
 
-Each scalar variable consumes 8 bytes of storage in memory.
-The reserved area in bank 0 from \$F700-\$FEFF can store 256 variables.
-Variables don't need to be declared, and their type is determined by an appended
-character. All variables without an appended character are regarded as REAL by default,
-and storage is claimed at their first usage. They are also initialised to zero,
-whereas string variables are initialised as an empty string "".
-
-All 104 one-letter variables are declared as {\bf fast} variables.
-26 user functions are declared as {\bf fast} functions.
-These are the variables (A - Z), (A\% - Z\%), (A\& - Z\&) and (A\$ - Z\$)
-and the functions (FNA() - FNZ()).
-They have fixed memory addresses in the range \$FD00 - \$FEFF, the address
-is generated by a hash algorithm from the variable name. The access to these
-variables and functions, either use or definition, is much faster,
-than the access to two letter variables and functions.
-The address of fast variables and functions is computed by a very fast algorithm,
-while the address of two-letter variables and functions is stored in a table,
-which has to be searched for every use.
+A program manipulates data by storing values in the computer's memory,
+referring to stored values, and updating them based on logic. In BASIC,
+elements of memory that store values are called {\em variables}. Each
+variable has a name, there are separate sets of variable names for each
+type of value.
+
+For example, the variable \screentext{AA} can store a number value. The
+variable \screentext{AA\$} can store a string value. Commodore BASIC
+considers these to be separate variables, even though the names both
+begin with \screentext{AA}.
+
+One way to store a value in a variable is with the assignment {\bf =}
+operator. For example:
+
+\begin{screenoutput}
+AA = 1.95
+
+AA$ = "HELLO, "
+\end{screenoutput}
+
+Variable names must start with a letter, and contain only letters and
+numbers. They can be of any length, but Commodore BASIC only recognizes
+the first two letters of the name. \screentext{SPEED} and \screentext{SP}
+would be considered the same variable.
+
+Variable names cannot contain any of the BASIC keywords. This makes using
+long names difficult: it is easy to use a keyword accidentally. For
+example, \screentext{ENFORCEMENT} is not a valid variable name, because
+{\bf FOR} is a keyword. It is common to use short variable names to avoid
+these hazards.
+
+Once assigned a value, a variable can be used within an expression with other
+constants, variables, functions, and operators:
+
+\begin{screenoutput}
+10 INPUT "WHAT IS YOUR NAME";NA$
+20 MSG$ = "HELLO, "+NA$+"!"
+20 PRINT MSG$
+\end{screenoutput}
+
+Unlike some programming languages, BASIC variables do not need to be declared
+before use. A variable has a default value of zero for number type variables,
+or the empty string (\screentext{""}) for string type variables.
+
+The variable types and their values are as follows.
```

Comment:
I noticed that later in Bit Shifter's old text, he was referring to his table of Byte/Integer/Real/String as the 'Scalar Variable Types'

I wonder if it's worth introducing them to the user this way too, to help them 'lock-in' the sense of which variables are 'deemed' to be scalar

I.e., changing this line to "The scalar variables types..."

It might mean moving up your description of a scalar variable, but let's see what you think

Reply:
Agreed, fixed. I think it reads well to move the scalar definition to the top of the variable section.